spec/nodeattr_interface_spec.rb

```diff
@@ -27,6 +27,23 @@
 require 'exceptions'
 
 RSpec.describe Metalware::NodeattrInterface do
+  context 'with setup1 genders' do
+    before do
+      SpecUtils.use_mock_genders(self, genders_file: 'setup1/genders')
+    end
+
+    describe '#nodes_in_primary_group' do
+      it 'returns only the nodes in the primary group' do
+        nodes = Metalware::NodeattrInterface.nodes_in_primary_group('group1')
+        nodes_not_in_primary_group = nodes.select do |node|
+          node.match(/\AnodeA\d\d\Z/).nil?
+        end
+        expect(nodes.length).to eq(10)
+        expect(nodes_not_in_primary_group.length).to eq(0)
```

Comment:
I'm not sure I follow - I don't mention anything about checking the length is shorter. I'm just saying that if you want to check a method returns the correct array, it seems the simplest and clearest way to do so would be a single assertion like `expect(nodes).to eq(['node01', 'node02', 'node03'])`. I can't see a downside to doing this in this case, and any checking of the lengths of the array or that each element matches a regex would then be redundant and only make things less clear.